internal/configuration/settings.go

```diff
@@ -179,6 +179,7 @@ func (s *Settings) Initialize() error {
 
 	certificates := certification.NewCertificates(s.Context, s.K8sClient)
 
+	// q. Why is this a separate step?
```

Comment:
separate from...?

Generally the idea of these Initialize functions are to load the data from the Kubernetes API before starting the watchers and waiting for the events to provide the data. There were some odd race conditions when the code worked that way.

In terms of this logic not being in the `NewXXXX` functions, that is a holdover from my C++ days where reaching outside the class for initialization was frowned upon.

Reply:
Oh yeah this was a note to myself that I'll remove.  Yeah I just saw a lot of code that was making sure `Initialize` was called before doing other things, and that the `Initialize` method didn't depend on other input itself so it just seemed like an extra thing the user of this module would have to remember to do.  Given that you have the safeguards in place at least it will be obviousl